File: Solutions.java/Two Pointers/0026. Remove Duplicates from Sorted Array.java
/// <summary>
/// 26. Remove Duplicates from Sorted Array
/// https://leetcode.com/problems/remove-duplicates-from-sorted-array
/// 
/// Difficulty Easy
/// Acceptance 56.9%
/// 
/// Array
/// Two Pointers
/// </summary>
class Solution
{
  public int removeDuplicates(int[] nums)
  {
    var n = 0;
    var prev = Integer.MAX_VALUE;

    for (var i = 0; i < nums.length; i++)
    {
      if (nums[i] != prev)
      {
        nums[n++] = nums[i];
        prev = nums[i];
      }
    }

    return n;
  }
}

File: Solutions.cs/Two Pointers/0026. Remove Duplicates from Sorted Array.cs
namespace Problem26
{

  /// <summary>
  /// 26. Remove Duplicates from Sorted Array
  /// https://leetcode.com/problems/remove-duplicates-from-sorted-array
  /// 
  /// Difficulty Easy
  /// Acceptance 56.9%
  /// 
  /// Array
  /// Two Pointers
  /// </summary>
  public class Solution
  {
    public int RemoveDuplicates(int[] nums)
    {
      if (nums.Length == 1)
        return 1;

      var k = 0;
      var i = 1;
      while (i < nums.Length)
      {
        if (nums[k] != nums[i])
        {
          k++;
          nums[k] = nums[i];
        }

        i++;
      }

      return k + 1;
    }
  }
}
